gsadjust/data_analysis.py

```diff
@@ -112,49 +112,31 @@ def compute_gravity_change(obstreemodel, table_type='simple'):
                                                           role=256)  # 256=QtCore.Qt.UserRole
                     # Iterate through, look for matching station. 'if' statements deal with Gravnet, which truncates
                     # station names to 6 characters
-                    if len(initial_station.station) > 6 and len(station_name) > 6:
-                        if initial_station.station == station_name:
-                            break
-                    elif len(initial_station.station) > 6:
-                        if initial_station.station[:6] == station_name:
-                            break
-                    elif initial_station.station == station_name:
+                    if (initial_station.station == station_name or initial_station.station[:6] == station_name):
                         break
-                    else:
-                        initial_station = None
+                else:
+                    # If we get to the end without breaking, set it to None.
+                    initial_station = None
```

Comment:
We can use the for-else construct here. It's a good option whenever your *success* path out of a loop is via `break`. The `else` runs only if the loop completes normally.